Ignore "folders" when processing s3 objects

From time to time we can see log lines similar to these:

```
INFO  [pool-7-thread-1] c.a.s.c.p.m.SqsManager - Downloaded log file PATH/TO/ACCOUNT/111222333444/ from our-cloudtrail-log-bucket
ERROR [pool-7-thread-1] c.a.s.c.p.i.DefaultExceptionHandler - Fail to parse log file.
com.amazonaws.services.cloudtrail.processinglibrary.exceptions.ProcessingLibraryException: Fail to parse log file.
at com.amazonaws.services.cloudtrail.processinglibrary.reader.EventReader.processSource(EventReader.java:163) [aws-cloudtrail-processing-library-1.0.2.jar!/:?]
at com.amazonaws.services.cloudtrail.processinglibrary.AWSCloudTrailProcessingExecutor$ScheduledJob$1.run(AWSCloudTrailProcessingExecutor.java:178) [aws-cloudtrail-processing-library-1.0.2.jar!/:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_66-internal]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_66-internal]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_66-internal]
Caused by: java.io.EOFException
at java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268) ~[?:1.8.0_66-internal]
at java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258) ~[?:1.8.0_66-internal]
at java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164) ~[?:1.8.0_66-internal]
at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79) ~[?:1.8.0_66-internal]
at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:91) ~[?:1.8.0_66-internal]
at com.amazonaws.services.cloudtrail.processinglibrary.reader.EventReader.processSource(EventReader.java:153) ~[aws-cloudtrail-processing-library-1.0.2.jar!/:?]
```

It seems like we get S3 notifications not only for new files, but also for each "folder" in our S3 bucket:
PATH/
PATH/TO/
PATH/TO/ACCOUNT/
PATH/TO/ACCOUNT/111222333444/
PATH/TO/ACCOUNT/111222333444/file.json.gzip

This happens due to our patch in the [fork](/zalando-stups/aws-cloudtrail-processing-library) of Amazon's Cloudtrail Processing Library, to support not only CloudTrail SNS notifications, but also S3 bucket SNS notifications.

We need to ignore all irrelevant S3 notifications, e.g. new folders.

--- a/fullstop-apis/fullstop-plugin-api/pom.xml
+++ b/fullstop-apis/fullstop-plugin-api/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-plugin-api</artifactId>

--- a/fullstop-apis/fullstop-violation-api/pom.xml
+++ b/fullstop-apis/fullstop-violation-api/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-violation-api</artifactId>

--- a/fullstop-aws-support/pom.xml
+++ b/fullstop-aws-support/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop-core/pom.xml
+++ b/fullstop-core/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-core</artifactId>

--- a/fullstop-it/pom.xml
+++ b/fullstop-it/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
 

--- a/fullstop-job-launcher/pom.xml
+++ b/fullstop-job-launcher/pom.xml
@@ -6,7 +6,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
 

--- a/fullstop-jobs/pom.xml
+++ b/fullstop-jobs/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-jobs</artifactId>

--- a/fullstop-logging/pom.xml
+++ b/fullstop-logging/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
 

--- a/fullstop-parent-poms/fullstop-parent/pom.xml
+++ b/fullstop-parent-poms/fullstop-parent/pom.xml
@@ -29,18 +29,18 @@
         <argLine />
         <license.skip>true</license.skip>
 
-        <spring-boot.version>1.3.0.RELEASE</spring-boot.version>
+        <spring-boot.version>1.3.1.RELEASE</spring-boot.version>
         <spring-security-oauth2.version>2.0.8.RELEASE</spring-security-oauth2.version>
-        <spring-cloud-starter-hystrix.version>1.0.3.RELEASE</spring-cloud-starter-hystrix.version>
+        <spring-cloud-starter-hystrix.version>1.0.4.RELEASE</spring-cloud-starter-hystrix.version>
         <archaius.version>0.7.3</archaius.version>
 
-        <aws-java-sdk.version>1.10.35</aws-java-sdk.version>
-        <aws-cloudtrail-processing-library.version>1.0.2</aws-cloudtrail-processing-library.version>
+        <aws-java-sdk.version>1.10.47</aws-java-sdk.version>
+        <aws-cloudtrail-processing-library.version>1.0.1-ZAL-2</aws-cloudtrail-processing-library.version>
 
-        <joda-time.version>2.9</joda-time.version>
+        <joda-time.version>2.9.1</joda-time.version>
         <commons-lang3.version>3.4</commons-lang3.version>
-        <guava.version>18.0</guava.version>
-        <json-path.version>2.0.0</json-path.version>
+        <guava.version>19.0</guava.version>
+        <json-path.version>2.1.0</json-path.version>
         <javax.el-api.version>2.2.5</javax.el-api.version>
 
         <maven-clean-plugin.version>3.0.0</maven-clean-plugin.version>
@@ -48,18 +48,18 @@
         <maven-scm-plugin.version>1.9.4</maven-scm-plugin.version>
         <maven-release-plugin.version>2.5.3</maven-release-plugin.version>
 
-        <querydsl.version>3.6.9</querydsl.version>
+        <querydsl.version>3.7.0</querydsl.version>
         <apt-maven-plugin.version>1.1.3</apt-maven-plugin.version>
         <jacoco-maven-plugin.version>0.7.5.201505241946</jacoco-maven-plugin.version>
 
-        <stups-spring-oauth2.version>1.0.5</stups-spring-oauth2.version>
-        <spring-boot-zalando-stups-tokens.version>0.9.3</spring-boot-zalando-stups-tokens.version>
-        <tokens.version>0.9.2</tokens.version>
+        <stups-spring-oauth2.version>1.0.8-RC1</stups-spring-oauth2.version>
+        <spring-boot-zalando-stups-tokens.version>0.9.5</spring-boot-zalando-stups-tokens.version>
+        <tokens.version>0.9.8</tokens.version>
         <kio-client.version>0.11.0</kio-client.version>
         <pierone-client.version>1.0.0</pierone-client.version>
         <kontrolletti-client-java.version>0.1.0-RC-1</kontrolletti-client-java.version>
 
-        <assertj-core.version>3.2.0</assertj-core.version>
+        <assertj-core.version>3.3.0</assertj-core.version>
         <mockito.version>1.10.19</mockito.version>
         <otj-pg-embedded.version>0.4.0</otj-pg-embedded.version>
     </properties>

--- a/fullstop-parent-poms/fullstop-parent/pom.xml
+++ b/fullstop-parent-poms/fullstop-parent/pom.xml
@@ -8,7 +8,7 @@
     </parent>
     <groupId>org.zalando.stups</groupId>
     <artifactId>fullstop-parent</artifactId>
-    <version>11-SNAPSHOT</version>
+    <version>11</version>
     <packaging>pom</packaging>
     <name>Fullstop -- The Build-Parent-Pom</name>
     <description>Fullstop parent -- Audit reporting application</description>
@@ -312,7 +312,7 @@
         <url>https://github.com/zalando-stups/fullstop.git</url>
         <connection>scm:git:git@github.com:zalando-stups/fullstop.git</connection>
         <developerConnection>scm:git:git@github.com:zalando-stups/fullstop.git</developerConnection>
-        <tag>HEAD</tag>
+        <tag>fullstop-parent-11</tag>
     </scm>
 
     <licenses>

--- a/fullstop-parent-poms/fullstop-parent/pom.xml
+++ b/fullstop-parent-poms/fullstop-parent/pom.xml
@@ -8,7 +8,7 @@
     </parent>
     <groupId>org.zalando.stups</groupId>
     <artifactId>fullstop-parent</artifactId>
-    <version>11</version>
+    <version>12-SNAPSHOT</version>
     <packaging>pom</packaging>
     <name>Fullstop -- The Build-Parent-Pom</name>
     <description>Fullstop parent -- Audit reporting application</description>
@@ -312,7 +312,7 @@
         <url>https://github.com/zalando-stups/fullstop.git</url>
         <connection>scm:git:git@github.com:zalando-stups/fullstop.git</connection>
         <developerConnection>scm:git:git@github.com:zalando-stups/fullstop.git</developerConnection>
-        <tag>fullstop-parent-11</tag>
+        <tag>HEAD</tag>
     </scm>
 
     <licenses>

--- a/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/PieroneTagProviderImplTest.java
+++ b/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/PieroneTagProviderImplTest.java
@@ -13,7 +13,7 @@ import java.util.Map;
 import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.util.Maps.newHashMap;
+import static com.google.common.collect.Maps.newHashMap;
 import static org.mockito.Mockito.*;
 
 public class PieroneTagProviderImplTest {

--- a/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/ScmSourceProviderImplTest.java
+++ b/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/ScmSourceProviderImplTest.java
@@ -11,7 +11,7 @@ import java.util.Map;
 import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.util.Maps.newHashMap;
+import static com.google.common.collect.Maps.newHashMap;
 import static org.mockito.Mockito.*;
 
 public class ScmSourceProviderImplTest {

--- a/fullstop-plugins/pom.xml
+++ b/fullstop-plugins/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop-testing/pom.xml
+++ b/fullstop-testing/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-testing</artifactId>

--- a/fullstop-violation/pom.xml
+++ b/fullstop-violation/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
     <artifactId>fullstop-violation</artifactId>

--- a/fullstop-web/fullstop-timemachine/pom.xml
+++ b/fullstop-web/fullstop-timemachine/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop-web/fullstop-violations-rest/pom.xml
+++ b/fullstop-web/fullstop-violations-rest/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop-web/fullstop-web-commons/pom.xml
+++ b/fullstop-web/fullstop-web-commons/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop-web/fullstop-web-test-support/pom.xml
+++ b/fullstop-web/fullstop-web-test-support/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath />
     </parent>
 

--- a/fullstop/pom.xml
+++ b/fullstop/pom.xml
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
 

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
 

--- a/team-service-api/pom.xml
+++ b/team-service-api/pom.xml
@@ -20,7 +20,7 @@
     <parent>
         <groupId>org.zalando.stups</groupId>
         <artifactId>fullstop-parent</artifactId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
     <modelVersion>4.0.0</modelVersion>

--- a/teams-service-client-spring/pom.xml
+++ b/teams-service-client-spring/pom.xml
@@ -5,7 +5,7 @@
     <parent>
         <artifactId>fullstop-parent</artifactId>
         <groupId>org.zalando.stups</groupId>
-        <version>10</version>
+        <version>11</version>
         <relativePath/>
     </parent>
     <modelVersion>4.0.0</modelVersion>

